R/database.R

```diff
@@ -61,38 +63,48 @@ get_ssm_by_sample = function(this_sample_id,
+  this_pair_status = pull(these_samples_metadata,pairing_status)
   if(verbose){
     print(paste("group:",this_unix_group,"genome:",this_genome_build))
   }
+  # get unmatched normal if necessary
+  if(pair_status == "unmatched"){
+    normal_sample_id = config::get("unmatched_normal_ids")[[this_unix_group]][[this_seq_type]][[this_genome_build]]
+    print(paste("NORMAL:",normal_sample_id))
+    return()
+  }
+  base_path=""
   if(flavour=="force_unmatched"){
     base_path = paste0(config::get("project_base"),this_unix_group,"/force_unmatched/",tool_name,"_vcf2maf_current/99-outputs/",seq_type,"--",this_genome_build)
   }else if(flavour=="augmented"){
     base_path = paste0(config::get("project_base"),this_unix_group,"/",tool_name,"_vcf2maf_current/level_3/augmented_mafs/99-outputs/",seq_type,"--",this_genome_build)
   }else if(flavour == "clustered"){
-    base_path = paste0(config::get("project_base"),this_unix_group,"/","slms_3-1.0_vcf2maf-1.3/99-outputs/",seq_type,"--",this_genome_build)
-  }else{
-    base_path = paste0(config::get("project_base"),this_unix_group,"/",tool_name,"_vcf2maf_current/99-outputs/",seq_type,"--",this_genome_build)
-  }
-  if(projection == "grch37"){
-    if(this_genome_build != projection){
-      maf_pattern = paste0(this_sample_id,"--*converted*maf")
-
+    if(deblacklisted){
+      base_path = paste0(config::get("project_base"),this_unix_group,"/","slms_3-1.0_vcf2maf-1.3/99-outputs/raw/")
     }else{
-      maf_pattern = paste0(this_sample_id,"--*final.maf")
 
+      path_template = config::get("results_filatfiles")$ssm$template$clustered
+      base_path = paste0(config::get("project_base"),this_unix_group,"/","slms_3-1.0_vcf2maf-1.3/99-outputs/deblacklisted/")
     }
-    maf_path = dir(base_path,pattern=glob2rx(maf_pattern))
-  }
-  if(length(maf_path)<1){
-    message(paste("NO FILE FOUND FOR THIS SAMPLE (flavour):",this_sample_id, "(",flavour,")"))
-    print(paste(base_path,maf_pattern))
-    return()
+  }else{
+    base_path = paste0(config::get("project_base"),this_unix_group,"/",tool_name,"_vcf2maf_current/99-outputs/",seq_type,"--",this_genome_build)
   }
-  if(length(maf_path)>1){
-    message(paste("TOO MANY FILES FOR THIS SAMPLE (flavour):",this_sample_id, "(",flavour,")"))
-    return()
+  maf_pattern = paste0(this_sample_id,"*",projection,".maf")
+  maf_path = paste0(base_path,"maf/",seq_type,"--projection/")
+  aug_maf_path = paste0(base_path,"augmented_maf/",seq_type,"--projection/")
+  maf_files = dir(maf_path,pattern=glob2rx(maf_pattern))
+  aug_maf_files = dir(aug_maf_path,pattern=glob2rx(maf_pattern))
+
+  if(augmented && length(aug_maf_files)){
+    full_maf_path = paste0(aug_maf_path,aug_maf_files)
+  }else{
+    #when augmented maf is missing, use the default
+    full_maf_path = paste0(maf_path,maf_files)
   }
-  sample_ssm = fread_maf(paste0(base_path,"/",maf_path))
+  print(paste("using:",full_maf_path))
+
+  sample_ssm = fread_maf(full_maf_path)
```

Comment:
I've tested the updated functions, get_gambl_metadata seems to run without errors but when running get_ssm_by_sample I get the following error:

` Error in data.table::fread(file = maf_file_path, sep = "\t", stringsAsFactors = FALSE,  : 
  File '/projects/nhl_meta_analysis_scratch/gambl/results_local/gambl/slms-3_vcf2maf_current/99-outputs/genome--hg38/BLGSP-71-06-00001-01A-11D--BLGSP-71-06-00001-99A-01D--matched_slms-3.final.converted_hg38ToHg19.maf' does not exist or is non-readable. getwd()=='/home/cmattsson/newtest/GAMBLR' `